playbooks/common/openshift-cluster/upgrades/docker/upgrade.yml

```diff
@@ -0,0 +1,44 @@
+---
+# We need docker service up to remove all the images, but these services will keep
+# trying to re-start and thus re-pull the images we're trying to delete.
+- name: Stop containerized services
+  service: name={{ item }} state=stopped
+  with_items:
+    - "{{ openshift.common.service_type }}-master"
+    - "{{ openshift.common.service_type }}-master-api"
+    - "{{ openshift.common.service_type }}-master-controllers"
+    - "{{ openshift.common.service_type }}-node"
+    - etcd_container
+    - openvswitch
+  failed_when: false
+  when: openshift.common.is_containerized | bool
```

Comment:
As far as actual implementation goes (for triggering upgrades via config.yml playbook runs), I would be thinking about having the new openshift_version role do comparisons of currently running version to the requested openshift_release (or image tag / pkg version) and check if this looks like the user asked for an upgrade. openshift_version would then set whatever facts we need to signal that an upgrade is underway, other roles would key off those, otherwise just protect installed versions.

i.e. docker role would detect that an upgrade is underway and assume to go to latest docker if this is a docker deployment. (unless docker_upgrade=False)

node role would detect an upgrade requested and handle all evac tasks.

When it comes to one off things like the steps required for image upgrade when crossing the Docker 1.10 boundary, we'll just have to embed that logic into the roles for as long as that is within the realm of reason for a supported path, and kill those tasks when it's not.